### PR TITLE
[WIP] Typed capability API

### PR DESCRIPTION
**Pull request by cedric-cordenier**

### CODE REVIEW

File: pkg/capabilities/v2/capability.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+	schema, err := jsonvalidate.CompileString("<uriPrefix>", string(schemab))
+	if err != nil {
+		return err
+	}
+	return schema.Validate(mapping)
+}
+
+func (c *capability[I, O, C]) Execute(ctx context.Context, callback chan<- capabilities.CapabilityResponse, request capabilities.CapabilityRequest) error {
+	tcb := make(chan CapabilityResponse[O])
```
Comment (HenryNguyen5) [inline pkg/capabilities/v2/capability.go:101]: What does `tcb` mean?

File: pkg/capabilities/v2/capability.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+			}
+
+			callback <- capabilities.CapabilityResponse{
+				Value: v,
+			}
+		}
+	}
+}
+
+func NewCapability[I, O, C any](cap Capability[I, O, C]) capabilities.CallbackExecutable {
```
Comment (HenryNguyen5) [inline pkg/capabilities/v2/capability.go:160]: This looks great, and is the direction I wanted to take the Validation API efforts to :)
What we're building represents a RPC framework of sorts.

 I'd imagine there's a lot of value to be had by allowing the user to inject middlewares that are called during request/response lifecycle

File: pkg/capabilities/v2/capability.go [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+)
+
+type CapabilityResponse[O any] struct {
+	Value O
+	Err   error
+}
+
+type CapabilityRequest[I, C any] struct {
```
Comment (bolekk) [inline pkg/capabilities/v2/capability.go:19]: How do those work in a remote setting? Does the receiver always know what type to expect or is there some type info encoded too?

Comment (HenryNguyen5) [reply pkg/capabilities/v2/capability.go:19]: I'd imagine it's the same, capabilities should have a input/config/output schema they abide to.

File: pkg/capabilities/v2/capability.go (hunk 2 above)
Comment (cedric-cordenier) [reply pkg/capabilities/v2/capability.go:160]: Yes agreed :) We could use this for observability and logging for example :)

File: pkg/capabilities/v2/capability.go (hunk 3 above)
Comment (cedric-cordenier) [reply pkg/capabilities/v2/capability.go:19]: @bolekk This is purely a client-side feature (i.e. inside the capability), anything accessing the capability (either via the Remote API or via the workflow engine would continue to do so via the lower-level, values-based API.

**Pull request closed without merging**